tools/loader/Loader.h

```diff
@@ -141,6 +167,18 @@ class Loader {
   /// from the run.
   void runInference(ExecutionContext *context, size_t batchSize = 1);
 
+  /// Register a loader extension.
+  Loader &registerExtension(std::unique_ptr<LoaderExtension> ext);
+  /// Called once after ONNX or Caffe2 model loading.
+  void postModelLoad(PlaceholderBindings &bindings, ProtobufLoader &protoLoader,
+                     size_t compilationBatchSize);
+  /// Called at the beginning of each mini-batch inference.
+  void inferInitMiniBatch(PlaceholderBindings &bindings, size_t minibatchIndex,
+                          size_t minibatchSize);
+  /// Called after the completion of each mini-batch inference.
+  void inferEndMiniBatch(PlaceholderBindings &, size_t minibatchIndex,
```

Comment:
I'm curious what exactly this one could be used for. Do you have any examples/ideas in mind, e.g. is this for processing results coming from a segmentation model or something? My first instinct was that this would be used as a replacement over `processAndPrintResults()` or something.

Reply:
Currently we are using Loader extension mainly for tensor instrumentation/dump and also for custom quantization profile generation.  `inferEndMiniBatch` is useful as soon as you want to scan the whole generated data and doing some reduction algorithm (the placeholder value of a minibatch execution will be lost when running the next mini-batch).